File: glip-team-messaging/glip-compliance-exports/create-data-export-task/code-samples/createDataExportTask.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// POST BODY
const body = {
    dateFrom: '<ENTER VALUE>',
    dateTo: '<ENTER VALUE>',
    userIds: [
        '<ENTER VALUE>',
    ],
    chatIds: [
        '<ENTER VALUE>',
    ]
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
await platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password });
const r = await platform.post(`/restapi/v1.0/glip/data-export`, body);

File: glip-team-messaging/glip-compliance-exports/create-data-export-task/code-samples/createDataExportTask.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// POST BODY
CreateDataExportTaskRequest createDataExportTaskRequest = new CreateDataExportTaskRequest {
    dateFrom = "<ENTER VALUE>",
    dateTo = "<ENTER VALUE>",
    userIds = new[] {
        "<ENTER VALUE>",
    },
    chatIds = new[] {
        "<ENTER VALUE>",
    }
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Glip().DataExport().Post(createDataExportTaskRequest);
// PROCESS RESPONSE
